### PR TITLE
Mepts 1073 tpt eligible changes

### PR DESCRIPTION
**Pull request by TinoMuianga**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java [hunk 1]
```diff
@@ -154,24 +154,6 @@ public CohortDefinition getTxCurrWithTPTCompletion() {
                 hivMetadata.getRestartConcept().getConceptId()),
             mapping));
 
-    compositionCohortDefinition.addSearch(
-        "B1",
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:158]: This is being used in the composition, any reason to remove it?

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
@TinoMuianga please have a look

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java (hunk 1 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:158]: Yes! For requirement 8 "Patients who already completed INH" it is asked to delete the whole B1 query hence I also removed it from the compositions where it is applied.

Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTCompletionCohortQueries.java:158]: B1 had two parts: B1part1(The one to delete) and B1part2, For TPTEligiblePatientListCohortQueries, yet for Completion which uses the same query, only called for B1. and this was giving an error when running the module.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 2]
```diff
@@ -505,53 +474,28 @@ public CohortDefinition getINHStartA1(
...
+            + "                         p.voided = 0 "
+            + "                   AND e.voided = 0 "
+            + "                   AND o.voided = 0 "
+            + "                   AND o2.voided = 0 "
+            + "                   AND o3.voided = 0 "
+            + "                   AND e.encounter_type = ${53} "
+            + "                   AND e.location_id = :location "
+            + "                   AND (o.concept_id = ${23985} AND o.value_coded = ${656}) "
+            + "                   AND (o2.concept_id = ${165308} AND o2.value_coded = ${1256}) "
+            + "                   AND (o3.concept_id = ${6128} AND o3.value_datetime IS NOT NULL "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: According to requirement the Isoniazid Prophylaxis Start Date (6128) obs must be removed, the INH start date must be checked on obs_datetime not on obs_value_datetime and the period must be 7 months not 210 days

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 3]
```diff
@@ -784,43 +671,46 @@ public CohortDefinition getINHStartA4(
...
+            + "                         pp.voided = 0"
+            + "                   AND ee.voided = 0"
+            + "                   AND o.voided = 0"
+            + "                   AND o2.voided = 0"
+            + "                   AND ee.encounter_type IN (${6}, ${9})"
+            + "                   AND ee.location_id = :location"
+            + "                   AND first_filt.patient_id = pp.patient_id "
+            + "                   AND (o.concept_id = ${23985} AND o.value_coded = ${656}) "
+            + "                   AND (o2.concept_id = ${165308} AND o2.value_coded = ${1256}) "
+            + "                   AND (o3.concept_id = ${6128} AND o3.value_datetime IS NOT NULL "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: according to spec the INH Start Date must be checked on 165308 obs datetime

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 4]
```diff
@@ -784,43 +671,46 @@ public CohortDefinition getINHStartA4(
...
+            + "                                 AND ( oo.concept_id = ${23985} AND oo.value_coded = ${656} ) "
             + "                                 AND oo.voided = 0 "
             + "                                 AND ee.voided = 0 "
-            + "                                 AND oo.value_coded = ${656} "
             + "                                 AND pp.voided = 0 "
             + "                                 AND ee.location_id = :location "
             + "                                 AND first_filt.patient_id = pp.patient_id "
-            + "                                 AND oo.value_datetime IS NOT NULL "
-            + "                                 AND oo.value_datetime >= Date_sub( "
+            + "                                 AND ( oo2.concept_id = ${6128} AND oo2.value_datetime IS NOT NULL "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: shouldn't it be removed?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 5]
```diff
@@ -721,7 +609,6 @@ public CohortDefinition getINHStartA4(
     map.put("23987", treatmentFollowUpTPTConcept);
     map.put("6128", dataInicioProfilaxiaIsoniazidaConcept);
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java]: shouldn't it be removed?

**Review by eliazmuchave — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 6]
```diff
@@ -1509,12 +1121,11 @@ public CohortDefinition getIPTB1part2() {
             + "UNION "
             + "SELECT p.patient_id FROM patient p INNER JOIN encounter e ON p.patient_id = e.patient_id "
             + "                                     INNER JOIN obs o ON e.encounter_id = o.encounter_id "
-            + "                                     INNER JOIN obs o2 ON e.encounter_id = o2.encounter_id "
-            + "WHERE p.voided =0 AND e.voided = 0 AND o.voided=0 AND o2.voided=0 "
+            + "WHERE p.voided =0 AND e.voided = 0 AND o.voided=0 "
             + "  AND e.encounter_type = ${53} "
             + "  AND o.concept_id = ${23985} AND o.value_coded = ${656} "
-            + "  AND o2.concept_id = ${6129} AND o2.value_datetime IS NOT NULL "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:1516]: The INH must be checked using the new TPT structure. Value_coded = Fim (id= 1267) for concept_id = 165308

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 7]
```diff
@@ -1526,12 +1137,11 @@ public CohortDefinition getIPTB1part2() {
...
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:1533]: The INH must be checked using the new TPT structure. Value_coded = Fim (id= 1267) for concept_id = 165308

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 8]
```diff
@@ -1543,12 +1153,11 @@ public CohortDefinition getIPTB1part2() {
...
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:1550]: The INH must be checked using the new TPT structure. Value_coded = Fim (id= 1267) for concept_id = 165308

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java [hunk 9]
```diff
@@ -3419,19 +2974,17 @@ public CohortDefinition get3HPLastProfilaxyDuringM3Period(sectionFromM section)
...
             + "                          AND e.voided = 0 "
             + "                          AND o.voided = 0 "
-            + "                          AND o2.voided = 0 "
             + "                          AND e.encounter_type = ${53} "
             + "                          AND (o.concept_id = ${23985} "
             + "                          AND o.value_coded = ${23954}) "
             + "                          AND e.location_id = :location "
             + "                          AND e.encounter_datetime <= :endDate "
-            + "                          AND ( o2.concept_id = ${6129} "
-            + "                          AND o2.value_datetime IS NOT NULL "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:3429]: should the end data verification not be verified using a new structure? Value_coded = Fim (id= 1267) for concept_id = 165308. Check the requirements

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java [hunk 10]
```diff
@@ -309,7 +217,8 @@ public CohortDefinition getPatientsWithUltimaProfilaxia3hp() {
             + "       AND e.location_id = :location "
             + "       AND e.encounter_type = ${53} "
             + "       AND ( o.concept_id = ${23985} AND o.value_coded = ${23954} ) "
-            + "       AND ( o2.concept_id = ${6128} AND o2.value_datetime BETWEEN :startDate AND :endDate ) ";
+            + "       AND ( o2.concept_id = ${6128} "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java]: Check the requirements.. should use new structure to check the 3HP Start Date. Valude_coded = Iniciar (id= 1256) for concept_id=165308 ObsDateTime for concept id = 165308

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java [hunk 11]
```diff
@@ -549,13 +457,13 @@ public CohortDefinition getPatientsWithRegimeDeTPT3HP() {
...
             + "                 AND ( o2.concept_id = ${23987} AND o2.value_coded IN ( ${1256}, ${1705} ) ) "
-            + "                 AND e.encounter_datetime BETWEEN :startDate AND :endDate) filt "
+            + "                 AND e.encounter_datetime BETWEEN DATE_SUB(:endDate, INTERVAL 4 MONTH) AND :endDate ) filt "
             + "                 ON filt.patient_id = p.patient_id "
-            + "           WHERE p.voided = 0 AND e.voided = 0 AND o6.voided = 0 AND o7.voided = 0 "
+            + "           WHERE p.voided = 0 AND e.voided = 0 AND o6.voided = 0 "
             + "           AND e.location_id = :location "
             + "           AND e.encounter_type = ${53} "
             + "           AND o6.concept_id = ${23985} AND o6.value_coded = ${23954} "
-            + "           AND o7.concept_id = ${6128} AND o7.value_datetime <= Date_sub(filt.start_date, interval 4 month) "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java:558]: The 3HP start date must be computed according to new TPT structure. Check the requirements

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TPTEligiblePatientsQueries.java [hunk 12]
```diff
@@ -416,28 +398,14 @@ public static String getY1QueryWithPatientIdForB5() {
...
-   * (concept id 6122) with value code “Inicio” (concept id 1256) or Profilaxia TPT (concept id
-   * 23985) value coded INH (concept id 656) and Estado da Profilaxia (concept id 165308) value
-   * coded Início (concept id 1256) and encounter datetime <= end date. or
+   * <p>Y2: Select Encounter Datetime from Ficha clinica (encounter type 6) with Profilaxia TPT
+   * (concept id 23985) value coded INH (concept id 656) and Estado da Profilaxia (concept id
+   * 165308) value coded Início (concept id 1256) and encounter datetime <= end date. or
    *
    * @return {@link String}
    */
   public static String getY2Query() {
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TPTEligiblePatientsQueries.java:413]: Where is the new implementation?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TPTEligiblePatientsQueries.java [hunk 13]
```diff
@@ -754,27 +717,17 @@ public static String getY6Query() {
...
-        + "                                AND ( o.concept_id = ${6122} "
-        + "                                    AND o.value_coded = ${1256} ) "
-        + "                               OR ( ( o2.concept_id = ${23985} "
-        + "                                AND o2.value_coded = ${656} ) "
-        + "                                AND ( o3.concept_id = ${165308} "
-        + "                                    AND o3.value_coded = ${1256} ) "
-        + "                                      ) "
-        + "                                AND e.encounter_datetime >= "
+        + "                                AND e.encounter_type IN ( ${6}, ${9} ) "
+        + "                                AND o.concept_id = ${6128} "
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TPTEligiblePatientsQueries.java]: Does not meet requirements

**Review by eliazmuchave — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java (hunk 10 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTInitiationCohortQueries.java:220]: Applied according To requirement

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 5 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:603]: Removed

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 4 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:676]: Removed

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java (hunk 3 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java:700]: checked on 165308 obs datetime.

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**